### PR TITLE
Lesson 6

### PR DESCRIPTION
**Pull request by oxtaco**

### CODE REVIEW

File: api/config.js [hunk 1]
```diff
@@ -1,4 +1,24 @@
-module.exports = {
-  PORT: 8080,
-  ttl_default: 2,
+const config = {
```
Comment (alexmaltsv) [inline api/config.js:1]: Better move all configuration to env files:  https://12factor.net/config

File: api/src/queue/route.js [hunk 2]
```diff
@@ -1,10 +1,8 @@
 const { Router } = require('express');
-const Controller = require('./controller');
+const controller = require('./init');
```
Comment (alexmaltsv) [inline api/src/queue/route.js:2]: init?

Comment (alexmaltsv) [reply api/src/queue/route.js:2]: typo. Check pls again

File: api/utils/adapt.js [hunk 3]
```diff
@@ -0,0 +1 @@
+module.exports = (req) => ({ ...req.body, ...req.params });
```
Comment (alexmaltsv) [inline api/utils/adapt.js:1]: Discussed this point. It's bad practice because there are possible collisions here

Comment (oxtaco) [reply api/utils/adapt.js:1]: it's not used anywhere, forgot to delete it again after merging

File: api/storage/redis/storageClient.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+const util = require('util');
+const StorageClientInterface = require('../storageClientInterface');
+
+module.exports = class StorageClient extends StorageClientInterface {
+  constructor(Database, DBname, type) {
+    super();
+    this.name = DBname;
+    this.type = type;
```
Comment (alexmaltsv) [inline api/storage/redis/storageClient.js:8]: Can u declare possible types and difference between this in comments?

File: api/storage/redis/storageClient.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+const util = require('util');
+const StorageClientInterface = require('../storageClientInterface');
+
+module.exports = class StorageClient extends StorageClientInterface {
+  constructor(Database, DBname, type) {
+    super();
+    this.name = DBname;
```
Comment (alexmaltsv) [inline api/storage/redis/storageClient.js:7]: In this case, good practice cover arguments for exceptions. 
Example: 

`if !(DBname) throw new SomeError()`

**Pull request merged**